### DISCUSSION TITLE
Save user's query

### DISCUSSION THREAD

**Discussion by redlinecodes** [appbaseio/reactivesearch#1751] [Q&A] [answered]
Is there a prop in DataSearch that can be used to send the user's selected or typed query to our database to compile user search history?

**Comment by siddharthlatest** [comment, marked as answer]
@redlinecodes You can create a side effect for this using the `onQueryChange` callback event. See the docs for the same: https://docs.appbase.io/docs/reactivesearch/v3/search/datasearch/#extending.
